### PR TITLE
Work around server-side issues with Accept-Language header handling

### PR DESCRIPTION
**Pull request by joewalsh**
Sets a single Wikipedia-specific language code to the `Accept-Language` header for zhwiki and srwiki. This works around server-side issues handling standard `Accept-Language` headers as documented in [T256491](https://phabricator.wikimedia.org/T256491). The `Accept-Language` header workaround can be removed when [T256491](https://phabricator.wikimedia.org/T256491) is fixed.

**Fixes Phabricator ticket:** 
https://phabricator.wikimedia.org/T254627

### Test Steps
Utilize repro steps from https://phabricator.wikimedia.org/T254627

### Screenshots

No preference:
No pref

Traditional:
Traditional

Simplified:
Simplified

### CODE REVIEW

**Comment by joewalsh** [comment]
Hold pending triage into 6.7.1 or later

File: Wikipedia/Code/ArticleFetcher.swift [hunk 1]
```diff
@@ -430,4 +431,66 @@ final public class ArticleFetcher: Fetcher, CacheFetching {
...
+                    completion(summary, response, nil)
+                case .failure(let error):
+                    completion(nil, response, error)
+                }
+            }
+        } catch let error {
+            completion(nil, nil, error)
+            return nil
+        }
+    }
```
Comment (joewalsh) [inline Wikipedia/Code/ArticleFetcher.swift:495]: Heads up that this isn't completely new code - it was brought in from the summary fetcher

File: Wikipedia/Code/ArticleFetcher.swift [hunk 2]
```diff
@@ -290,8 +290,9 @@ final public class ArticleFetcher: Fetcher, CacheFetching {
     }
     
     public func urlRequest(from url: URL, cachePolicy: WMFCachePolicy? = nil, headers: [String: String] = [:]) -> URLRequest? {
-        
-        let request = urlRequestFromPersistence(with: url, persistType: .article, cachePolicy: cachePolicy, headers: headers)
+        var requestHeaders = configuration.pageContentServiceHeaders(for: url.wmf_language)
```
Comment (tonisevener) [inline Wikipedia/Code/ArticleFetcher.swift:293]: Only thing I've noticed is I think the Edit Preview endpoints need these `Accept-Language` values set as well. It doesn't look like they build their `URLRequest`s by leaning on this method.

**Review by tonisevener — CHANGES_REQUESTED** [review]

**Comment by tonisevener** [comment]
I noticed this wasn't working in the Staging environment - it wasn't properly pulling `wmf_language` for that url form. I attempted to fix in [this commit](https://github.com/wikimedia/wikipedia-ios/commit/b91d1e0f61c339719c5831cfd15aff260994b53d). Feel free to pull it in if you'd like, it felt a bit messy though so I didn't want to commit directly.

**Review by tonisevener — APPROVED** [review]

**Pull request merged**